agent/config/runtime_test.go

```diff
@@ -3223,120 +3223,6 @@ func TestConfigFlagsAndEdgecases(t *testing.T) {
 			}`},
 			err: "config_entries.bootstrap[0]: invalid name (\"invalid-name\"), only \"global\" is supported",
 		},
-		{
-			desc: "ConfigEntry bootstrap proxy-defaults (snake-case)",
```

Comment:
This test case is `TestConfigFlagsAndEdgecases`. The tests fail because the hcl and json produce different opaque config in the `Config` section.  I think with this change these are no longer special edgecases. The standard behaviour is handled by the TestFullConfig test case. 

I couldn't think of any test to replace these, but I am happy to write some more if anyone has ideas.